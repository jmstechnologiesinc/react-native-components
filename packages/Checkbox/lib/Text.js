"use strict";
var __assign = (this && this.__assign) || function () {
    __assign = Object.assign || function(t) {
        for (var s, i = 1, n = arguments.length; i < n; i++) {
            s = arguments[i];
            for (var p in s) if (Object.prototype.hasOwnProperty.call(s, p))
                t[p] = s[p];
        }
        return t;
    };
    return __assign.apply(this, arguments);
};
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
var __rest = (this && this.__rest) || function (s, e) {
    var t = {};
    for (var p in s) if (Object.prototype.hasOwnProperty.call(s, p) && e.indexOf(p) < 0)
        t[p] = s[p];
    if (s != null && typeof Object.getOwnPropertySymbols === "function")
        for (var i = 0, p = Object.getOwnPropertySymbols(s); i < p.length; i++) {
            if (e.indexOf(p[i]) < 0 && Object.prototype.propertyIsEnumerable.call(s, p[i]))
                t[p[i]] = s[p[i]];
        }
    return t;
};
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
var jsx_runtime_1 = require("react/jsx-runtime");
var React = __importStar(require("react"));
var react_native_1 = require("react-native");
var types_1 = require("./types");
var LightTheme_1 = __importDefault(require("./LightTheme"));
// @component-group Typography
/**
 * Typography component showing styles complied with passed `variant` prop and supported by the type system.
 *
 * <div class="screenshots">
 *   <img class="small" src="screenshots/typography.png" />
 * </div>
 *
 * ## Usage
 * ```js
 * import * as React from 'react';
 * import { Text } from 'react-native-paper';
 *
 * const MyComponent = () => (
 *   <>
 *     <Text variant="displayLarge">Display Large</Text>
 *     <Text variant="displayMedium">Display Medium</Text>
 *     <Text variant="displaySmall">Display small</Text>
 *
 *     <Text variant="headlineLarge">Headline Large</Text>
 *     <Text variant="headlineMedium">Headline Medium</Text>
 *     <Text variant="headlineSmall">Headline Small</Text>
 *
 *     <Text variant="titleLarge">Title Large</Text>
 *     <Text variant="titleMedium">Title Medium</Text>
 *     <Text variant="titleSmall">Title Small</Text>
 *
 *     <Text variant="bodyLarge">Body Large</Text>
 *     <Text variant="bodyMedium">Body Medium</Text>
 *     <Text variant="bodySmall">Body Small</Text>
 *
 *     <Text variant="labelLarge">Label Large</Text>
 *     <Text variant="labelMedium">Label Medium</Text>
 *     <Text variant="labelSmall">Label Small</Text>
 *  </>
 * );
 *
 * export default MyComponent;
 * ```
 *
 * @extends Text props https://reactnative.dev/docs/text#props
 */
var Text = function (_a, ref) {
    var _b, _c;
    var style = _a.style, variant = _a.variant, rest = __rest(_a, ["style", "variant"]);
    var root = React.useRef(null);
    // FIXME: destructure it in TS 4.6+
    var writingDirection = react_native_1.I18nManager.isRTL ? 'rtl' : 'ltr';
    React.useImperativeHandle(ref, function () { return ({
        setNativeProps: function (args) { var _a; return (_a = root.current) === null || _a === void 0 ? void 0 : _a.setNativeProps(args); },
    }); });
    if (LightTheme_1.default.isV3 && variant) {
        var stylesByVariant = Object.keys(types_1.MD3TypescaleKey).reduce(function (acc, key) {
            var _a;
            var _b = LightTheme_1.default.typescale[key], fontSize = _b.fontSize, fontWeight = _b.fontWeight, lineHeight = _b.lineHeight, letterSpacing = _b.letterSpacing, fontFamily = _b.fontFamily;
            return __assign(__assign({}, acc), (_a = {}, _a[key] = __assign(__assign({}, (react_native_1.Platform.OS === 'android' && { fontFamily: fontFamily })), { fontSize: fontSize, fontWeight: fontWeight, lineHeight: lineHeight, letterSpacing: letterSpacing, color: LightTheme_1.default.colors.onSurface }), _a));
        }, {});
        var styleForVariant = stylesByVariant[variant];
        return ((0, jsx_runtime_1.jsx)(react_native_1.Text, __assign({ ref: root, style: [styleForVariant, styles.text, { writingDirection: writingDirection }, style] }, rest)));
    }
    else {
        return ((0, jsx_runtime_1.jsx)(react_native_1.Text, __assign({}, rest, { ref: root, style: [
                __assign(__assign({}, (!LightTheme_1.default.isV3 && ((_b = LightTheme_1.default.fonts) === null || _b === void 0 ? void 0 : _b.regular))), { color: LightTheme_1.default.isV3 ? (_c = LightTheme_1.default.colors) === null || _c === void 0 ? void 0 : _c.onSurface : LightTheme_1.default.colors.text }),
                styles.text,
                style,
            ] })));
    }
};
var styles = react_native_1.StyleSheet.create({
    text: {
        textAlign: 'left',
    },
});
exports.default = React.forwardRef(Text);
