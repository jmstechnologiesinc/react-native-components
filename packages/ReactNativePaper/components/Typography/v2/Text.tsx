import * as React from 'react';
import {
  Text as NativeText,
  TextStyle,
  StyleProp,
  StyleSheet,
} from 'react-native';

import theme from '../../../styles/themes/v3/LightTheme';

type Props = React.ComponentProps<typeof NativeText> & {
  style?: StyleProp<TextStyle>;
  /**
   * @optional
   */
};

// @component-group Typography

/**
 * Text component which follows styles from the theme.
 *
 * @extends Text props https://reactnative.dev/docs/text#props
 */
const Text: React.ForwardRefRenderFunction<{}, Props> = (
  {style, ...rest}: Props,
  ref,
) => {
  const root = React.useRef<NativeText | null>(null);

  React.useImperativeHandle(ref, () => ({
    setNativeProps: (args: Object) => root.current?.setNativeProps(args),
  }));

  return (
    <NativeText
      {...rest}
      ref={root}
      style={[
        {
          ...(!theme.isV3 && theme.fonts?.regular),
          color: theme.isV3 ? theme.colors?.onSurface : theme.colors.text,
        },
        styles.text,
        style,
      ]}
    />
  );
};

const styles = StyleSheet.create({
  text: {
    textAlign: 'left',
  },
});

export default React.forwardRef(Text);
