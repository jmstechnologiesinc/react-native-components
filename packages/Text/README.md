# `Text`

> TODO: description

## Usage

```
const text = require('Text');

// TODO: DEMONSTRATE API
```
