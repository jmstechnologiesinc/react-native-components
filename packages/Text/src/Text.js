export {default} from '../../ReactNativePaper/components/Typography/Text';
