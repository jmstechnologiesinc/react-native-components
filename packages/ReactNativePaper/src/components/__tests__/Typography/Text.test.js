import * as React from 'react';
import renderer from 'react-test-renderer';
import Text from '../../Typography/Text.tsx';

it('renders every variant of Text with children as content', () => {
  const content = 'Something rendered as a child content';

  const variants = (
    <>
      <Text variant="displayLarge">{content}</Text>
      <Text variant="displayMedium">{content}</Text>
      <Text variant="displaySmall">{content}</Text>

      <Text variant="headlineLarge">{content}</Text>
      <Text variant="headlineMedium">{content}</Text>
      <Text variant="headlineSmall">{content}</Text>

      <Text variant="titleLarge">{content}</Text>
      <Text variant="titleMedium">{content}</Text>
      <Text variant="titleSmall">{content}</Text>

      <Text variant="bodyLarge">{content}</Text>
      <Text variant="bodyMedium">{content}</Text>
      <Text variant="bodySmall">{content}</Text>

      <Text variant="labelLarge">{content}</Text>
      <Text variant="labelMedium">{content}</Text>
      <Text variant="labelSmall">{content}</Text>
    </>
  );

  const tree = renderer.create(variants).toJSON();

  expect(tree).toMatchSnapshot();
});
