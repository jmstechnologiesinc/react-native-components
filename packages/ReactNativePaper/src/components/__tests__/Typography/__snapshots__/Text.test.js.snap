// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders every variant of Text with children as content 1`] = `
Array [
  <Text
    style={
      Array [
        Object {
          "color": "rgba(28, 27, 31, 1)",
          "fontSize": 57,
          "fontWeight": "400",
          "letterSpacing": 0,
          "lineHeight": 64,
        },
        Object {
          "textAlign": "left",
        },
        Object {
          "writingDirection": "ltr",
        },
        undefined,
      ]
    }
  >
    Something rendered as a child content
  </Text>,
  <Text
    style={
      Array [
        Object {
          "color": "rgba(28, 27, 31, 1)",
          "fontSize": 45,
          "fontWeight": "400",
          "letterSpacing": 0,
          "lineHeight": 52,
        },
        Object {
          "textAlign": "left",
        },
        Object {
          "writingDirection": "ltr",
        },
        undefined,
      ]
    }
  >
    Something rendered as a child content
  </Text>,
  <Text
    style={
      Array [
        Object {
          "color": "rgba(28, 27, 31, 1)",
          "fontSize": 36,
          "fontWeight": "400",
          "letterSpacing": 0,
          "lineHeight": 44,
        },
        Object {
          "textAlign": "left",
        },
        Object {
          "writingDirection": "ltr",
        },
        undefined,
      ]
    }
  >
    Something rendered as a child content
  </Text>,
  <Text
    style={
      Array [
        Object {
          "color": "rgba(28, 27, 31, 1)",
          "fontSize": 32,
          "fontWeight": "400",
          "letterSpacing": 0,
          "lineHeight": 40,
        },
        Object {
          "textAlign": "left",
        },
        Object {
          "writingDirection": "ltr",
        },
        undefined,
      ]
    }
  >
    Something rendered as a child content
  </Text>,
  <Text
    style={
      Array [
        Object {
          "color": "rgba(28, 27, 31, 1)",
          "fontSize": 28,
          "fontWeight": "400",
          "letterSpacing": 0,
          "lineHeight": 36,
        },
        Object {
          "textAlign": "left",
        },
        Object {
          "writingDirection": "ltr",
        },
        undefined,
      ]
    }
  >
    Something rendered as a child content
  </Text>,
  <Text
    style={
      Array [
        Object {
          "color": "rgba(28, 27, 31, 1)",
          "fontSize": 24,
          "fontWeight": "400",
          "letterSpacing": 0,
          "lineHeight": 32,
        },
        Object {
          "textAlign": "left",
        },
        Object {
          "writingDirection": "ltr",
        },
        undefined,
      ]
    }
  >
    Something rendered as a child content
  </Text>,
  <Text
    style={
      Array [
        Object {
          "color": "rgba(28, 27, 31, 1)",
          "fontSize": 22,
          "fontWeight": "400",
          "letterSpacing": 0,
          "lineHeight": 28,
        },
        Object {
          "textAlign": "left",
        },
        Object {
          "writingDirection": "ltr",
        },
        undefined,
      ]
    }
  >
    Something rendered as a child content
  </Text>,
  <Text
    style={
      Array [
        Object {
          "color": "rgba(28, 27, 31, 1)",
          "fontSize": 16,
          "fontWeight": "500",
          "letterSpacing": 0.15,
          "lineHeight": 24,
        },
        Object {
          "textAlign": "left",
        },
        Object {
          "writingDirection": "ltr",
        },
        undefined,
      ]
    }
  >
    Something rendered as a child content
  </Text>,
  <Text
    style={
      Array [
        Object {
          "color": "rgba(28, 27, 31, 1)",
          "fontSize": 14,
          "fontWeight": "500",
          "letterSpacing": 0.1,
          "lineHeight": 20,
        },
        Object {
          "textAlign": "left",
        },
        Object {
          "writingDirection": "ltr",
        },
        undefined,
      ]
    }
  >
    Something rendered as a child content
  </Text>,
  <Text
    style={
      Array [
        Object {
          "color": "rgba(28, 27, 31, 1)",
          "fontSize": 16,
          "fontWeight": "400",
          "letterSpacing": 0.15,
          "lineHeight": 24,
        },
        Object {
          "textAlign": "left",
        },
        Object {
          "writingDirection": "ltr",
        },
        undefined,
      ]
    }
  >
    Something rendered as a child content
  </Text>,
  <Text
    style={
      Array [
        Object {
          "color": "rgba(28, 27, 31, 1)",
          "fontSize": 14,
          "fontWeight": "400",
          "letterSpacing": 0.25,
          "lineHeight": 20,
        },
        Object {
          "textAlign": "left",
        },
        Object {
          "writingDirection": "ltr",
        },
        undefined,
      ]
    }
  >
    Something rendered as a child content
  </Text>,
  <Text
    style={
      Array [
        Object {
          "color": "rgba(28, 27, 31, 1)",
          "fontSize": 12,
          "fontWeight": "400",
          "letterSpacing": 0.4,
          "lineHeight": 16,
        },
        Object {
          "textAlign": "left",
        },
        Object {
          "writingDirection": "ltr",
        },
        undefined,
      ]
    }
  >
    Something rendered as a child content
  </Text>,
  <Text
    style={
      Array [
        Object {
          "color": "rgba(28, 27, 31, 1)",
          "fontSize": 14,
          "fontWeight": "500",
          "letterSpacing": 0.1,
          "lineHeight": 20,
        },
        Object {
          "textAlign": "left",
        },
        Object {
          "writingDirection": "ltr",
        },
        undefined,
      ]
    }
  >
    Something rendered as a child content
  </Text>,
  <Text
    style={
      Array [
        Object {
          "color": "rgba(28, 27, 31, 1)",
          "fontSize": 12,
          "fontWeight": "500",
          "letterSpacing": 0.5,
          "lineHeight": 16,
        },
        Object {
          "textAlign": "left",
        },
        Object {
          "writingDirection": "ltr",
        },
        undefined,
      ]
    }
  >
    Something rendered as a child content
  </Text>,
  <Text
    style={
      Array [
        Object {
          "color": "rgba(28, 27, 31, 1)",
          "fontSize": 11,
          "fontWeight": "500",
          "letterSpacing": 0.5,
          "lineHeight": 16,
        },
        Object {
          "textAlign": "left",
        },
        Object {
          "writingDirection": "ltr",
        },
        undefined,
      ]
    }
  >
    Something rendered as a child content
  </Text>,
]
`;
